client/core/core_test.go

```diff
@@ -2763,18 +2787,40 @@ func TestTradeTracking(t *testing.T) {
 	}
 	tBtcWallet.badSecret = false
 
+	tBtcWallet.redeemErrChan = make(chan error, 1)
+	rig.db.updateMatchChan = make(chan order.MatchStatus, 1)
+	defer func() {
+		rig.db.updateMatchChan = nil
+		tBtcWallet.redeemErrChan = nil
+	}()
 	rig.ws.queueResponse(msgjson.RedeemRoute, redeemAcker)
 	err = handleRedemptionRoute(tCore, rig.dc, msg)
 	if err != nil {
 		t.Fatalf("redemption message error: %v", err)
 	}
-	checkStatus("taker complete", order.MatchComplete)
+	err = <-tBtcWallet.redeemErrChan
+	if err != nil {
+		t.Fatalf("should have worked, got: %v", err)
+	}
+	// For taker there's one status update to MakerRedeemed on bcast...
+	status := <-rig.db.updateMatchChan
+	if status != order.MakerRedeemed {
+		t.Fatalf("wrong match status wanted %v, got %v", order.MakerRedeemed, status)
+	}
+	// and another to MatchComplete when the 'redeem' request back to the server
+	// succeeds.
+	status = <-rig.db.updateMatchChan
+	if status != order.MatchComplete {
+		t.Fatalf("wrong match status wanted %v, got %v", order.MatchComplete, status)
+	}
 	if !bytes.Equal(proof.MakerRedeem, redemptionCoin) {
 		t.Fatalf("redemption coin ID not logged")
 	}
 	if len(proof.TakerRedeem) == 0 {
 		t.Fatalf("taker redemption not sent")
 	}
+	rig.db.updateMatchChan = nil
+	tBtcWallet.redeemErrChan = nil
```

Comment:
I don't think you need the `defer func` above then.

Reply:
The defer was in case this test fails above before getting here, so that subsequent test funcs don't have these chans set.

Reply:
Unlike some of the server packages, the "rig" isn't reused in **core** tests.